src/backer/cardaning.py

```diff
@@ -59,44 +65,104 @@ def __init__(self, name='backer', hab=None, ks=None):
         balance = self.getaddressBalance(self.spending_addr.encode())
         if balance and balance > MINIMUN_BALANCE:
             print("Address balance:", balance/1000000, "ADA")
-        else:
+        else:            
             self.fundAddress(self.spending_addr)
 
+    def updateTip(self, tipHeight):        
+        self.tipHeight = tipHeight
+
     def publishEvent(self, event: bytes):
         self.pending_kel = event + self.pending_kel
 
-    def flushQueue(self):
+    def submitKelTx(self, kel):        
         try:
-            if self.pending_kel:
+            if kel:
                 # Build transaction
                 builder = pycardano.TransactionBuilder(self.context)
                 builder.add_input_address(self.spending_addr)
                 builder.add_output(pycardano.TransactionOutput(self.spending_addr, pycardano.Value.from_primitive([TRANSACTION_AMOUNT])))
-                self.pending_kel = bytes(self.pending_kel)
+                kel_data = bytes(kel)
                 # Chunk size
                 # bytearrays is not accept
-                value = [self.pending_kel[i:i + 64] for i in range(0, len(self.pending_kel), 64)]
+                value = [kel_data[i:i + 64] for i in range(0, len(kel_data), 64)]
 
                 # Metadata. accept int key type
                 builder.auxiliary_data = pycardano.AuxiliaryData(pycardano.Metadata({1: value}))
 
                 signed_tx = builder.build_and_sign([self.payment_signing_key],
                                                 change_address=self.spending_addr,
                                                 merge_change=True)
+                
                 # Submit transaction
-                self.context.submit_tx(signed_tx.to_cbor())  
-                self.pending_kel = bytearray()
-        except Exception as e:    
-            logger.critical(f"Error: {e}")
+                self.context.submit_tx(signed_tx.to_cbor())
+                transId = str(signed_tx.transaction_body.inputs[0].transaction_id)
+                trans = {
+                    "id": transId,
+                    "kel": kel.decode('utf-8'),
+                    "publish_time": help.toIso8601()
+                }
+                self.keldbConfirming.pin(keys=transId, val=json.dumps(trans).encode('utf-8'))
+        except Exception as e:
+            logger.critical(f"Cannot submit transaction: {e}")
+
+    def emptyPendingKEL(self):
+        self.pending_kel = bytearray()
+
+    def flushQueue(self):
+        if self.pending_kel is not None and self.pending_kel != b"":
+            self.submitKelTx(self.pending_kel)        
+            self.pending_kel = bytearray()
+
+    def slotToTime(self, slot):
+        timeStamp = SHELLY_UNIX + slot
```

Comment:
Cool, so ogmios doesn't provide this detail as it's too low level? Can you point to documentation on this?

Reply:
I havenot found any field for the time in Ogmios Trans [here](https://gitlab.com/viperscience/ogmios-python/-/blob/main/src/ogmios/model/cardano_model.py?ref_type=heads#L2053)
So I have to follow this [links ](https://github.com/ada9000/PythonCardanoSlotsScript/blob/main/CardanoSlots.py)to convert.

Reply:
OK, but shelley time will be different on all networks. Maybe we can simplify and say if it doesn't appear after X blocks? Instead of time. That would probably make more sense actually.

Reply:
So will we change this AC?
>If a transaction does not appear after 30mins, the KEL entries should be re-published on-chain (env variable)
to
>If a transaction does not appear after X blocks , the KEL entries should be re-published on-chain (env variable)

Reply:
updated the AC. Lets do it